src/smt/theory_str_noodler/aut_assignment.cpp

```diff
@@ -158,4 +158,34 @@ namespace smt::noodler {
             cur_level = next_level;
         }
     }
+
+    bool AutAssignment::is_flat(const BasicTerm& t) const {
+        bool flat = true;
+
+        mata::nfa::Nfa aut = *this->at(t);
```

Comment:
Should this be a (const) reference? Otherwise we are copying the automaton.
```suggestion
        const mata::nfa::Nfa& aut = *this->at(t);
```